Cookbook/Technical-Documents/DebuggingPushNotifications.md

```diff
@@ -0,0 +1,37 @@
+
+Each application that supports push notifications has a separate configuration on PubNub. Depending on your access level you may have access to all the apps (admin) or to specific apps. All the apps are grouped under `steven.hamblin` account.
+
+Each of the apps configurations contains multiple `key set`s for each of our backend environments, i.e. `dev` is a key set used on our dev environment, and so on. These key sets contain some private API keys that our notifications service uses to communicate with Pub Nub API and they are different for each environment.
+
+Each key set defines configuration for Pub Nub feautures, including push notifications. They contain FCM key (for Android, so ignore it) and push notifications certificate and APNS environment. For all of our non production environments we use enterprise certificate. For production environment we use AppStore certificate as that's the only environment that can be used in AppStore builds and we don't run any enterprise (Hockeyapp) builds against production environment.
+
+Apart from certificate we need to set `APNS Environment`. This is not related to _our_ environments (dev, preprod or prod) but as the name of this setting says to APNS itself. It is possible to set this to Development or Production. Production environment is used to deliver notifications to the apps distributed through Hockeyapp, Testflight or AppStore. Development environment is used to send notifications to the apps deployed to device locally with Xcode.
+
+For iOS we have separate key sets for push and VoIP notifications because PubNub as of Sept 2019 didn't yet support using single certificate for both types of notifications. Key sets for push notifications are prefixed with `ios-push` and are associated with regular push notifications certificate instead of VoIP certificate. All the ceritificates for all the apps can be found in 1Password iOS vault.
+
+## TLDR
+When you are testing push notifications make sure that:
+
+- the app is running against exepcted backend environment
+- PubNub configuration for this environment set to Production APNS Environment
+	- open the key set for this environment
+	- tap on "Replace  Certificate" button
+	- find the correct certificate PEM file in 1Password iOS vault and upload it
+		- for non production environment use Enterprise certificate
+		- for production environment only use AppStore certificate (or simply just never change production key set)
+	- select proper APNS environment:
+		- for local builds (made from Xcode) select Development environment
+		- for Hockeyapp builds select Production environment
+- make sure you notify in ios slack channel that you are changing PubNub environment
+- remember to change this configuration back when you are done and notificy ios channel
+- you are on Babylon-Partners WiFi network (West Office)
```

Comment:
Why do we need to be on this Wifi? It all works from home as well for me.

Reply:
In the office it was reported few times that some wifi network used for some QA is not suitable. Don't know what's special about it though.

Reply:
Then you can put, check that you are not on that QA WiFi. Otherwise, a remote person will think that push notifications are not working for him/her because he/she are not on correct WiFi

Reply:
```suggestion
- you are on Babylon-Partners WiFi network (if you are in the West Office)
```